x/ratelimit/client/cli/cli_test.go

```diff
@@ -0,0 +1,318 @@
+				fmt.Sprintf("--%s=%s", flags.FlagBroadcastMode, flags.BroadcastBlock),
+				strideclitestutil.DefaultFeeString(s.cfg),
+			},
+			quota,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+
+		s.Run(tc.name, func() {
+			cmd := cli.CmdAddRateLimit()
+			args := make([]string, len(tc.args))
+			copy(args, tc.args)
+			args[2] = "10"
+			args[3] = "20"
+			clientCtx := val.ClientCtx
+
+			// Add a new ratelimit
+			_, err := clitestutil.ExecTestCLICmd(clientCtx, cmd, args)
+			s.Require().NoError(err)
+
+			// Update the rate limit
+			cmd = cli.CmdUpdateRateLimit()
+			_, err = clitestutil.ExecTestCLICmd(clientCtx, cmd, tc.args)
+			s.Require().NoError(err)
+
+			// Check if ratelimit was updated properly
+			cmd = cli.GetCmdQueryRateLimit()
+			out, err := clitestutil.ExecTestCLICmd(clientCtx, cmd, []string{
+				path.Denom,
+				path.ChannelId,
+				fmt.Sprintf("--%s=json", tmcli.OutputFlag),
+			})
+			s.Require().NoError(err)
+
+			var result types.RateLimit
+			s.Require().NoError(clientCtx.Codec.UnmarshalJSON(out.Bytes(), &result))
+			s.Require().Equal(tc.expQuota, *result.Quota)
+		})
+	}
+}
+
+func (s *IntegrationTestSuite) TestCmdTxResetRateLimit() {
+	val := s.network.Validators[0]
+	quota := types.Quota{
+		MaxPercentSend: 20,
+		MaxPercentRecv: 30,
+		DurationHours:  1,
+	}
+	path := types.Path{
+		Denom:     sdk.DefaultBondDenom,
+		ChannelId: "channel-2",
+	}
+	flow := types.Flow{
+		Inflow:       0,
+		Outflow:      0,
+		ChannelValue: s.cfg.StakingTokens.Uint64(),
+	}
+
+	testCases := []struct {
+		name    string
+		args    []string
+		expFlow types.Flow
+	}{
+		{
+			"reset-rate-limit tx",
+			[]string{
+				sdk.DefaultBondDenom, // denom
+				path.ChannelId,       // channelId
+				fmt.Sprintf("--%s=json", tmcli.OutputFlag),
+				fmt.Sprintf("--%s=%s", flags.FlagFrom, val.Address.String()),
+				// common args
+				fmt.Sprintf("--%s=true", flags.FlagSkipConfirmation),
+				fmt.Sprintf("--%s=%s", flags.FlagBroadcastMode, flags.BroadcastBlock),
+				strideclitestutil.DefaultFeeString(s.cfg),
+			},
+			flow,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+
+		s.Run(tc.name, func() {
+			cmd := cli.CmdAddRateLimit()
+			clientCtx := val.ClientCtx
+
+			// Add a new ratelimit
+			_, err := clitestutil.ExecTestCLICmd(clientCtx, cmd, []string{
+				sdk.DefaultBondDenom,                    // denom
+				path.ChannelId,                          // channelId
+				fmt.Sprintf("%d", quota.MaxPercentSend), // maxPercentSend
+				fmt.Sprintf("%d", quota.MaxPercentRecv), // maxPercentRecv
+				fmt.Sprintf("%d", quota.DurationHours),  // durationHours
+				fmt.Sprintf("--%s=json", tmcli.OutputFlag),
+				fmt.Sprintf("--%s=%s", flags.FlagFrom, val.Address.String()),
+				// common args
+				fmt.Sprintf("--%s=true", flags.FlagSkipConfirmation),
+				fmt.Sprintf("--%s=%s", flags.FlagBroadcastMode, flags.BroadcastBlock),
+				strideclitestutil.DefaultFeeString(s.cfg),
+			})
+			s.Require().NoError(err)
+
```

Comment:
is there a way to send a transfer message here so the flow can go:
1. Add rate limit
2. Transfer 
3. Check that flow matches transfer
4. Reset
5. Check that flow is 0

?

Reply:
Great point - totally agree!